Framework/Algorithms/test/CorelliPowderCalibrationDatabaseTest.h

```diff
@@ -0,0 +1,431 @@
+    TS_ASSERT_DELTA(dupsampletable->cell<double>(0, 2), -0.0002, 0.000001);
+
+    // Clean
+    AnalysisDataService::Instance().remove(outwsname);
+  }
+
+  //-----------------------------------------------------------------------------
+  /**
+   * @brief Test main features required by the algorithm
+   * CorelliPowderCalibrationDatabase
+   */
+  void test_exec() {
+
+    // Create the test environment
+    // create directory
+    std::string calibdir{"TestCorelliPowderCalibration1117"};
+    // clean previous
+    boost::filesystem::remove_all(calibdir);
+    // create data base
+    boost::filesystem::create_directory(calibdir);
+    // create a previously generated database file
+    std::vector<std::string> banks{"source", "sample", "bank2", "bank42"};
+    create_existing_database_files(calibdir, banks);
+
+    // Create workspaces
+    EventWorkspace_sptr input_ws = createTestEventWorkspace();
+    // Name of the output calibration workspace
+    std::string outwsname(
+        "CorelliPowderCalibrationDatabaseTest_CombinedTableWS");
+    TableWorkspace_sptr calib_ws =
+        createTestCalibrationTableWorkspace(outwsname);
+    TS_ASSERT(input_ws);
+    TS_ASSERT(calib_ws);
+
+    // Init algorithm
+    CorelliPowderCalibrationDatabase alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+
+    // Set up
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", input_ws));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("InputCalibrationPatchWorkspace", calib_ws));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("DatabaseDirectory", calibdir));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setPropertyValue("OutputWorkspace", outwsname));
+
+    // Execute
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    // Verifying results
+    // Output 3: the combined calibration workspace
+    TS_ASSERT(AnalysisDataService::Instance().doesExist(outwsname));
+    TableWorkspace_sptr combinedcalibws =
+        std::dynamic_pointer_cast<TableWorkspace>(
+            AnalysisDataService::Instance().retrieve(outwsname));
+    TS_ASSERT(combinedcalibws);
+    // shall be 5 components
+    TS_ASSERT_EQUALS(combinedcalibws->rowCount(), 5);
+    TS_ASSERT_EQUALS(combinedcalibws->cell<std::string>(1, 0), "sample");
+    TS_ASSERT_EQUALS(combinedcalibws->cell<std::string>(2, 0), "bank1");
+    TS_ASSERT_EQUALS(combinedcalibws->cell<std::string>(4, 0), "bank42");
+
+    // Output 2: search the saved output calibration file
+    boost::filesystem::path pdir(calibdir);
+    boost::filesystem::path pbase("corelli_instrument_20201117.csv");
+    boost::filesystem::path ptodaycalfile = pdir / pbase;
+    std::string todaycalfile = ptodaycalfile.string();
+    TS_ASSERT(boost::filesystem::exists(todaycalfile));
+    // load and compare
+    // ... ...
+
+    // Output 1: check all the files
+    std::vector<std::string> compnames{"source", "sample", "bank1", "bank2",
```

Comment:
to be consistent, can you use CORELLI's component names for source and sample ("moderator" and "sample-position")?